docs/getting_started/developer-guide.md

```diff
@@ -0,0 +1,248 @@
+# Developer Guide
+
+This developer guide includes more complex interactions like
+contributing registry entries and building containers. If you haven't
+read `getting_started-installation`{.interpreted-text role="ref"} you
+should do that first. If you want to see a more general user guide, look
+at the [oras.land Python
+guide](https://oras.land/client_libraries/1_python/).
+
+## Documentation
+
+The documentation is provided in the `docs` folder of the repository,
+and generally most content that you might want to add is under
+`getting_started`. For ease of contribution, files that are likely to be
+updated by contributors (e.g., mostly everything but the module generated files)
+ are written in markdown. If you need to use [toctree](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#table-of-contents) you should not use extra newlines or spaces (see index.md files for exampls).
+Markdown is the chosen language for the oras community, and this is why we chose to
+use it over restructured syntax - it makes it easier to contribute documentation.
+
+
+### Install Dependencies and Build
+
+The documentation is built using sphinx, and generally you can install
+dependencies:
+
+``` console
+# in main oras-py folder (oras-py is needed to build docs)
+$ pip install -e .
+
+# Now docs dependencies
+cd docs
+pip install -r requrements.txt
+
+# Build the docs into _build/html
+make html
+```
+
+### Preview Documentation
+
+After `make html` you can cd into `_build/html` and start a local web
```

Comment:
How about using` enter into` instead of `cd into` here?

Reply:
I think (personally speaking) “enter into” reads more strangely than “cd into” because I’m used to the cd command.

Reply:
But if you feel strongly about it I can change it - I’m probably not the most representative user reading the docs!

Reply:
okay this is changed.